### PR TITLE
fix!: adjust UiList to design

### LINKED ISSUE #20: [REFACTOR] Adjust UiList to design
Components should be in accordance with the design.
* UiListItem with borders according to design
* UiListItem with padding according to design (mobile 12px 20px; desktop 12px) 
Zrzut ekranu 2022-10-18 o 15 09 26
https://www.figma.com/file/54rgvRJfBBagt4F34rrp1s/Core-Component-Library?node-id=2651%3A67786
* UiListItem with Button in Suffix according to design.
Zrzut ekranu 2022-10-18 o 15 15 36
* UiListItem with hover, active background according to design.
* UiListItem with `ui-list-item--has-error` according to design.
Zrzut ekranu 2022-10-18 o 15 17 57
* Add story for UiList as Answer With Radio 3 State
* Add story for UiList as Condition

### PR DESCRIPTION
**Pull request by ArkadiuszKrzymienski**
# Related issue
Close #20 

# Scope of work
Adjust UiList component to design, after changes: 
- UiListItem has borders and padding
Zrzut ekranu 2022-10-31 o 08 23 27

- UiListItem has button `UiListItemSuffix` component
Zrzut ekranu 2022-10-31 o 08 26 42

- UiListItem has error state and  hover, active background
Zrzut ekranu 2022-10-31 o 08 29 51

- Add story for UiList as multiple Choices
Zrzut ekranu 2022-10-31 o 08 28 19
Zrzut ekranu 2022-10-31 o 08 33 27

- Add story for UiList as condition 
Zrzut ekranu 2022-10-31 o 08 28 47

### CODE REVIEW

File: src/components/organisms/UiList/UiList.vue [hunk 1]
```diff
@@ -32,30 +40,29 @@
...
-  name: string;
-  text: string;
+export interface ListItemAttrs {
+  icon?: Icon;
+  hasSuffix?: boolean;
+  suffixAttrs?: ListItemSuffixAttrs;
+  [key: string]: unknown;
 }
-export interface ListRenderItemWithChildren {
+export interface ListItemAsObj {
```
Comment (AlanLes) [inline src/components/organisms/UiList/UiList.vue:60]: Or maybe `ListItemComplex`, wdyt? :)

File: src/components/organisms/UiList/_internal/UiListItem.vue [hunk 2]
```diff
@@ -3,32 +3,133 @@
...
 <script setup lang="ts">
+import { computed } from 'vue';
 import type { PropType } from 'vue';
 import type { HTMLTag } from '../../../../types/tag';
+import type { Icon } from '../../../../types/icon';
+import type { PropsAttrs } from '../../../../types/attrs';
+import UiListItemSuffix from './UiListItemSuffix.vue';
 
-defineProps({
+const props = defineProps({
```
Comment (AlanLes) [inline src/components/organisms/UiList/_internal/UiListItem.vue:33]: What do you think about **[this](https://vuejs.org/api/sfc-script-setup.html#typescript-only-features)** pattern with `withDefaults`? Wouldn't it be cleaner than all of those casts `as PropType<HTMLTag>,`, `as PropType<Icon>,`, `Object as PropsAttrs,` etc? :) 
+ It seems like vue3 docs suggest to rather use `withDefaults` 🤔

**Review by AlanLes — APPROVED** [review]
Looks great! As always @ArkadiuszKrzymienski, great job! 👍 🙂  
I've found one type's name which i'd rename. 
I also noticed that common pattern all over the project is `defineProps` with casting to `PropType<Type>`, but recently I've found out that vue-docs suggest to rather use `withDefaults` instead of all of those casts 😄 So am I. I'd reconsider using that recommended pattern 🙂

File: src/components/organisms/UiList/_internal/UiListItem.vue (hunk 2 above)
Comment (ArkadiuszKrzymienski) [reply src/components/organisms/UiList/_internal/UiListItem.vue:33]: oh cool 😀 earlier it was probably experimental. I would try this pattern. What do you think @pspaczek?

File: src/components/organisms/UiList/UiList.vue (hunk 1 above)
Comment (ArkadiuszKrzymienski) [reply src/components/organisms/UiList/UiList.vue:60]: yeah great, thanks 👍

File: src/components/organisms/UiList/_internal/UiListItemSuffix.vue [hunk 3]
```diff
@@ -0,0 +1,81 @@
+<template>
+  <UiButton class="ui-button--text ui-button--small ui-list-item-suffix">
+    <!-- @slot Use this slot to replace label template. -->
+    <slot
+      name="label"
+      v-bind="{ label }"
+    >
+      {{ label }}
```
Comment (pspaczek) [inline src/components/organisms/UiList/_internal/UiListItemSuffix.vue:8]:
```suggestion
      <span v-bind="labelAttrs">{{ label }}</span>
```
I find 1 use case when we need that version. When you have a button without label you should add it
* as `aria-label` to button 
* add `visual-hidden` to label, and that, why we need this span. WDYT?

File: src/components/organisms/UiList/_internal/UiListItemSuffix.vue [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+      name="icon"
+      v-bind="{
+        hasIcon,
+        iconSuffixAttrs: defaultProps.iconAttrs
+      }"
+    >
+      <UiIcon
+        v-if="icon"
+        v-bind="defaultProps.iconAttrs"
+        class="ui-list-item-suffix__icon ui-button__icon "
```
Comment (pspaczek) [inline src/components/organisms/UiList/_internal/UiListItemSuffix.vue:21]:
```suggestion
        class="ui-button__icon ui-button__icon--right ui-list-item-suffix__icon "
```

File: src/components/organisms/UiList/_internal/UiListItemSuffix.vue [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+const hasIcon = computed(() => !!defaultProps.value.iconAttrs.icon);
+</script>
+
+<style lang="scss">
+@use "../../../../styles/functions";
+
+.ui-list-item-suffix {
+  $element: list-item-suffix;
+
+  --button-icon-margin: #{functions.var($element + "-icon", margin, 0)};
```
Comment (pspaczek) [inline src/components/organisms/UiList/_internal/UiListItemSuffix.vue:76]:
```suggestion
```

File: src/components/organisms/UiList/_internal/UiListItemSuffix.vue [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+  display: flex;
+  gap: functions.var($element, gap, var(--space-4));
```
Comment (pspaczek) [inline src/components/organisms/UiList/_internal/UiListItemSuffix.vue:79]:
```suggestion
```
IMO if we want to made some changes here we should make it in UiButton :). In this case you should use `ui-button__icon--right`

Comment (pspaczek) [reply src/components/organisms/UiList/_internal/UiListItemSuffix.vue:79]: Probably I miss this part in UiMenuItemSuffix 🙈

File: src/components/organisms/UiList/_internal/UiListItemSuffix.vue (hunk 5 above)
Comment (pspaczek) [reply src/components/organisms/UiList/_internal/UiListItemSuffix.vue:76]: and you miss  `--button-rtl-icon-margin` 👍

**Review by pspaczek — CHANGES_REQUESTED** [review]
@ArkadiuszKrzymienski I found a few improvements to UiList, UiListItem, and UiListItemSuffix. I did not check the other changes. Maybe you can close this PR and split it into 2 PRs 
1. UiList changes
2. updates in the components that use UiList WDYT?

File: src/components/organisms/UiList/UiList.vue [hunk 7]
```diff
@@ -9,19 +9,27 @@
         v-for="(item, key) in itemsToRender"
         :key="key"
       >
-        <UiListItem v-bind="(item as ListRenderItemWithChildren).listItemAttrs">
+        <UiListItem
+          v-bind="item"
+          class="ui-list__item"
```
[the remark this answers is not in the document]
Comment (ArkadiuszKrzymienski) [reply src/components/organisms/UiList/UiList.vue:14]: we don't need this 🙈 thanks 🙂

File: src/components/organisms/UiList/_internal/UiListItemSuffix.vue (hunk 3 above)
Comment (ArkadiuszKrzymienski) [reply src/components/organisms/UiList/_internal/UiListItemSuffix.vue:8]: nice, thanks 😀

**Pull request merged**